inspirehep/modules/authors/views.py

```diff
@@ -154,6 +154,79 @@ def get_inspire_url(data):
     return url
 
 
+@blueprint.route('/publications', methods=['GET'])
+def get_publications():
+    from elasticsearch.helpers import scan
+    from invenio_search import current_search_client
+
+    recid = request.values.get('recid', 0, type=int)
+
+    publications = []
+    collaborations = set()
+    keywords = set()
+
+    for result in scan(
+            current_search_client,
+            query={
+                '_source': ['collaboration',
+                            'earliest_date',
+                            'publication_info',
+                            'titles',
+                            'thesaurus_terms'],
+                'query': {"match": {"authors.recid": recid}}
+            },
+            index='records-hep',
+            doc_type='hep'):
+
+        try:
+            result_source = result['_source']
+            publication = {}
+
+            # Get publication title.
+            publication['title'] = result_source.get(
+                'titles', {})[0].get('title')
+
+            # Get journal title.
+            try:
+                publication['journal'] = result_source.get(
+                    'publication_info', {})[0].get(
+                    'journal_title')
+            except TypeError:
+                pass
+
+            # Get publication year.
+            try:
+                publication['year'] = result_source.get(
+                    'publication_info', {})[0].get(
+                    'year')
+            except TypeError:
+                pass
+
+            # Get keywords.
+            for keyword in result_source.get('thesaurus_terms', []):
+                if keyword.get('keyword') is not "* Automatic Keywords *" \
+                        and keyword.get('keyword'):
+                    keywords.add(keyword.get('keyword'))
+
+            # Get collaborations.
+            for collaboration in result_source.get('collaboration', []):
+                collaborations.add(collaboration.get('value'))
+
+            if publication:
+                publications.append(publication)
+
+        except KeyError:
```

Comment:
This seems like a too big `try...except` block. You can just add it around the `title` code.